ministryofjustice/analytics-platform-control-panel
# Allow users to authenticate with Azure in order to capture their Justice email address

## :memo: Summary
This PR contributes to https://github.com/ministryofjustice/data-platform/issues/3605

This PR implements a new page that allows a user to authenticate with Azure/EntraID using their @justice.gov.uk email address. This is intended as a one time process, in order to capture their justice email address.

After login, users are redirected to the page where it explains the purpose of authenticating. There is a button to skip the process, which allows users to continue as normal. The next time they log in they will be redirected to the new page again. I have set an arbritary deadline of 30th April, but this is a WIP and has no real effect.

![image](https://github.com/ministryofjustice/analytics-platform-control-panel/assets/15347726/1e47b2f1-b676-4ba6-b88b-9fed94cb8d04)

![image](https://github.com/ministryofjustice/analytics-platform-control-panel/assets/15347726/b3a6ac41-a46c-4971-8470-8958c7bac3f5)

The changes in this PR are needed as part of our wider work around offering Quicksight to AP users via the new AP UI.

## :mag: What should the reviewer concentrate on?
- The text displayed to the user - this is a WIP/first draft so any suggestions about the messaging here are welcome
- Authentication process

## :technologist: How should the reviewer test these changes?
NOTE: Ideally you should log out of your Justice identity (e.g. outlook) first. By doing this you will see the full flow. If you complete the following steps while already logged in, it will still work and you will capture your @justice.gov.uk email but you wont go through the full steps of logging in with your Justice email.
- You will need to add some additional env vars (ask me)
- Run the project locally
- Log in as usual
- You should be redirected to the new page
- Click the "Skip for now" button. This should allow you to look at tools, datasources etc
- Log out and back in
- You should be redirected to the new page - click "Authenticate" button.
- You will be redirected to log in with your justice email, as you would for logging in to any MS service (such as Outlook).
- After completing the process, you are redirected and a success message is displayed showing your @justice.gov.uk email
- Log out and back in again - you should NOT be redirected to the new page this time, as your details are already stored.

## :books: Documentation status
